src/app/common/base.people.component.ts

```diff
@@ -262,11 +262,10 @@ export abstract class BasePeopleComponent<UserType extends ProviderUserUserDetai
                 childComponent.publicKey = publicKey;
                 childComponent.onConfirmedUser.subscribe(async () => {
                     try {
-                        await confirmUser(publicKey);
+                        childComponent.formPromise = confirmUser(publicKey);
+                        await childComponent.formPromise;
                         this.modal.close();
-                    } catch (e) {
-                        this.logService.error(`Handled exception: ${e}`);
-                    }
+                    } catch { }
```

Comment:
Hmm you're right. I don't think I processed that this is just logging the error in console. That seems fine to me.

I took it out because the typical flow for the api action directive handles errors for us, but it doesn't log.

Reply:
You know what, I think the right thing to do is to log the error in the directive.

Reply:
I think that's a good call for sure. Can we delete the try/catch block here, if the promise error handling is done by the directive?

Reply:
The later.

We still need the try/catch or we'll throw here as well. Promises are a little weird in that anybody who has the gall to wait on them, has to deal with the error from a rejection. Since the directive is now totally handling it for us, we just need to subdue the error. It's a common pattern in the codebase, but not one that I love. If you have ideas around it that would be a great clean-up.

Reply:
Yeah I suspected the answer might be something like that, just wasn't 100% on how the directive + component would interact there.